File: node/src/automation.js
const axios = require("axios");

class Automation {
  base_url = "https://api.go-mailer.com";

  constructor(api_key) {
    this.api_key = api_key;
  }

  async triggerEvent(event_code, user_email, context = {}) {
    if (!event_code) throw new Error("Event code is required.");
    if (!user_email) throw new Error("User email is required.");
    if (!Object.keys(context).length) throw new Error("Context is required. Please provide relevant context data.");

    const body = { event_code, contact_email: user_email, context };
    const { data: response } = await axios.post(`${this.base_url}/v1/automations`, body, {
      headers: { Authorization: `Bearer ${this.api_key}` },
    });

    return response;
  }
}

module.exports = { Automation };

File: python/gomailer/automation.py
import requests


class Automation:
    base_url = 'https://api.go-mailer.com'

    def __init__(self, api_key):
        self.api_key = api_key

    def trigger_event(self, event_code, contact_email, context={}):
        if not event_code:
            raise Exception('Event code is required.')
        elif not contact_email:
            raise Exception('Contact email is not required.')
        elif not bool(context):
            raise Exception('No context data provided.')

        body = {
            'event_code': event_code,
            'contact_email': contact_email,
            'context': context
        }
        headers = {'Authorization': f'Bearer {self.api_key}'}

        response = requests.post(
            f"{self.base_url}/v1/automations'", json=body, headers=headers)
        return response
